File: Android/AndesiteV9/src/xbee/regular/AtRemoteResponse.java
package xbee.regular;

import java.util.ArrayList;

public class AtRemoteResponse {
	
	public static String getCommand(ArrayList<Byte> packet)
	{
		char first = (char)(byte)packet.get(15);
		char second = (char)(byte)packet.get(16);
		return String.valueOf(first) + String.valueOf(second);
	}
	
	public static String getFrameId(ArrayList<Byte> packet)
	{
		return Common.byteToHex(packet.get(4));
	}
	
	public static String getStatus(ArrayList<Byte> packet)
	{
		return Common.byteToHex(packet.get(17));
	}
	
	public static String getData(ArrayList<Byte> packet)
	{
		if(packet.size() > 19)
		{
			int start = 18;
			int length = packet.size() - 1 - start;
			byte[] temp = new byte[length];
			for(int n = 0; n < length; n++)
			{
				temp[n] = packet.get(n + start);
			}
			return Common.byteToHex(temp);
		}
		
		return null;
	}
	
	public static String getSh(ArrayList<Byte> packet)
	{
		int length = 4;
		int start = 5;
		
		byte[] temp = new byte[length];
		for(int n = 0; n < length; n++)
		{
			temp[n] = packet.get(n + start);
		}
		return Common.byteToHex(temp);
	}
	
	public static String getSl(ArrayList<Byte> packet)
	{
		int length = 4;
		int start = 9;
		
		byte[] temp = new byte[length];
		for(int n = 0; n < length; n++)
		{
			temp[n] = packet.get(n + start);
		}
		return Common.byteToHex(temp);
	}
	
	public static String getMy(ArrayList<Byte> packet)
	{
		int length = 2;
		int start = 13;
		
		byte[] temp = new byte[length];
		for(int n = 0; n < length; n++)
		{
			temp[n] = packet.get(n + start);
		}
		return Common.byteToHex(temp);
	}
	
	public static String getNi(ArrayList<Byte> packet)
	{
		int length = packet.size() - 1;
		byte temp[] = new byte[length - 18];
		String ni = "";
		
		for(int x = 18; x < length; x++)
		{
			temp[x - 18] = packet.get(x);
			ni += String.valueOf((char)Integer.parseInt(packet.get(x).toString()));
		}
		
		return ni;
	}

}

File: Arduino/Arduino Libraries/Version 9 Code/split/Xbee2/util/AtRemoteResponse.cpp
#include "Arduino.h"
#include "AtRemoteResponse.h"

byte AtRemoteResponse::getFrameID(byte packet[])
{
	return packet[FRAME_ID_INDEX];
}

String AtRemoteResponse::getCommand(byte packet[])
{
	char first = packet[COMMAND_INDEX];
	char second = packet[COMMAND_INDEX];
	return first + second;
}

byte AtRemoteResponse::getStatus(byte packet[])
{
	return packet[STATUS_INDEX];
}

void AtRemoteResponse::getData(byte data[], byte packet[], int packetLength)
{
	if(packetLength > 9) {
		int length = packetLength - 1 - DATA_INDEX;
		for(int i = 0; i < length; i++) {
			data[i] = packet[DATA_INDEX + i];
		}
	}
}

long AtRemoteResponse::getSH(byte packet[])
{
	int length = 4;
	long SH;

	for(int i = 0; i < length; i++) {
		SH += (packet[SH_INEDX + i] << (24 - (8 * i)));
	}
	return SH;
}

long AtRemoteResponse::getSL(byte packet[])
{
	int length = 4;
	long SL;

	for(int i = 0; i < length; i++) {
		SL += (packet[SL_INEDX + i] << (24 - (8 * i)));
	}
	return SL;
}

int AtRemoteResponse::getMy(byte packet[])
{
	int length = 2;
	long MY;

	for(int i = 0; i < length; i++) {
		MY += (packet[MY_INEDX + i] << (8 - (8 * i)));
	}
	return MY;
}